core/lib/zksync_core/src/consensus/mod.rs

```diff
@@ -1,61 +1,212 @@
 //! Consensus-related functionality.
-use std::sync::Arc;
-
 use anyhow::Context as _;
-use zksync_concurrency::{ctx, scope};
-use zksync_consensus_executor::{ConsensusConfig, Executor, ExecutorConfig};
+use zksync_concurrency::{ctx, error::Wrap as _, scope};
+use zksync_consensus_executor as executor;
 use zksync_consensus_roles::{node, validator};
-use zksync_dal::ConnectionPool;
+use zksync_consensus_storage::BlockStore;
+use zksync_dal::{consensus_dal::Payload, ConnectionPool};
 use zksync_types::Address;
 
-mod payload;
-mod proto;
+use crate::sync_layer::sync_action::ActionQueueSender;
+
 mod storage;
 
 #[cfg(test)]
 pub(crate) mod testonly;
 #[cfg(test)]
 mod tests;
 
-pub(crate) use self::{payload::Payload, storage::sync_block_to_consensus_block};
+use std::collections::{HashMap, HashSet};
+
+use serde::de::Error;
+use zksync_consensus_crypto::{Text, TextFmt};
+
+use self::storage::Store;
```

Comment:
Please place all `use`s before module definitions so that they are appropriately grouped by the formatter.